python/rascaline/transformer.py

```diff
@@ -0,0 +1,296 @@
+    a new TensorMap, thus saving computation time and maintaining a single
+    representation for all representations.
+
+    Parameters:
+    -----------
+    :param selectors: The selector to be used when selecting the features.
+        Currently the same selector is used for different blocks.
+        #TODO: It should be possible for the user to pass a list of selectors
+        for blocks.
+    :param transformation: the type of transformation to be performed.
+        Two options are possible - 'keys_to_features' and 'keys_to_properties'.
+        #TODO: provide the ability to pass a list of conversions that will occur
+        one after the other.
+    :param moved_keys: Those keys which will be moved during the transformation.
+        This variable can accept knowledge of type str (one key), list (a list
+        of keys) and Labels (in addition to the name of the keys, pass a list
+        of keys to be moved).
+    """
+
+    def __init__(self, selector, transformation=None, moved_keys=None):
+        #
+        self.selector = selector
+        self.transformation = transformation
+        self.moved_keys = moved_keys
+        if (
+            (self.transformation is not None)
+            and (self.transformation != "keys_to_samples")
+            and (self.transformation != "keys_to_properties")
+        ):
+            raise ValueError(
+                "`transformation` parameter should be either `keys_to_samples`,"
+                f" either `keys_to_properties`, got {self.transformation}"
+            )
+        if (self.transformation is None) and (self.moved_keys is not None):
+            raise ValueError("unable to shift keys: unknown transformation type")
+
+    def _copy(self, tensor_map):
+        """This function that allows you to create a copy of 'TensorMap'.
+        It may be worth adding this function to the equistore.
+        """
+        blocks = []
+        for _, block in tensor_map:
+            blocks.append(block.copy())
+        return TensorMap(tensor_map.keys, blocks)
+
+    def keys_definition(self):
+        """This is another internal function that performs two main tasks.
+        First, it converts all moved_keys to the same format.  What is
+        meant is that further we need the names of the keys we are going
+        to move, as well as the 'TensorMap' keys, which will be passed
+        to the compute function as a reference at the end. This function
+        stores the names of the moved keys in the 'moved_keys_names' array,
+        and stores the keys of the final TensorMap reference in 'final_keys'.
+        """
+        # the first 2 cases are simple - we either copy the moved_keys directly,
+        # or create an array based on them, and simply take all the keys passed
+        # in the fit TensorMap step as the final keys.
+        if isinstance(self.moved_keys, str):
+            self.moved_keys_names = [self.moved_keys.copy()]
```

Comment:
copy is not needed, python strings are immutable

```suggestion
            self.moved_keys_names = [self.moved_keys]
```